diff --git a/Practica2/driver/imx477_mode_tbls.h b/Practica2/driver/imx477_mode_tbls.h
index 3978e42..795bbd5 100644
--- a/Practica2/driver/imx477_mode_tbls.h
+++ b/Practica2/driver/imx477_mode_tbls.h
@@ -42,6 +42,7 @@ static const imx477_reg imx477_mode_common[] = {
 	/* software reset */
 	{0x0103, 0x01},
 	{IMX477_TABLE_WAIT_MS, IMX477_WAIT_MS*10},
+	{0x0101, 0x02}, 
 	{0x0136, 0x18},
 	{0x0137, 0x00},
 	{0x0808, 0x02},
@@ -356,6 +357,7 @@ static const imx477_reg imx477_mode_3840x2160_30fps[] = {
 	{0x0342, 0x2B},
 	{0x0343, 0xC0},
 	{0x0340, 0x09},
+	{0x0101, 0x02}, 
 	{0x0341, 0xC4},
 	{0x0344, 0x00},
 	{0x0345, 0x00},
@@ -473,6 +475,7 @@ static const imx477_reg imx477_mode_1920x1080_60fps[] = {
 	{0x0114, 0x01},
 	{0x0342, 0x1B},
 	{0x0343, 0x58},
+	{0x0101, 0x02},
 	{0x0340, 0x07},
 	{0x0341, 0xD0},
 	{0x0344, 0x00},
